app/controllers/questions_controller.rb

```diff
@@ -32,6 +32,26 @@ def create
 
     respond_to do |format|
       if @question.save
+        # check if there are any tags in the params for the question
+        if params[:tags_list]
+          # get names of tags the user wants to add to the question
+          tag_names_array = params[:tags_list].split(",")
+          # empty question's tag array.
+          @question.tags = []
+          # loop through names from user input.
+          tag_names_array.each do |tagName|
+            # check if tag with this tagName already exists
+            if Tag.exists?(tag: tagName)
```

Comment:
Much better!
I think you can use this: https://chodounsky.net/2015/09/14/create-record-if-it-does-not-exist-in-activerecord/

This way, you don't need the if/else check here. Looks like you can just do `@t = Tag.find_or_create_by(tag: "John",tagName)`

Lastly now that this has been simplified, I think you can kill the comments around this area.